src/js/components/SharedComponents/DropdownTypeahead.jsx

```diff
@@ -0,0 +1,350 @@
+  */
+
+import React, { PropTypes } from 'react';
+import _ from 'lodash';
+import Awesomplete from 'awesomplete';
+import DropdownTypeaheadCheckbox from './DropdownTypeaheadCheckbox';
+
+import TypeaheadWarning from './TypeaheadWarning';
+
+const propTypes = {
+  bubbledRemovedFilterValue: PropTypes.shape({
+    filter: PropTypes.string,
+    value: PropTypes.shape({
+      userId: PropTypes.number,
+      name: PropTypes.string,
+    }),
+  }),
+  formatter: PropTypes.func,
+  onSelect: PropTypes.func.isRequired,
+  internalValue: PropTypes.array,
+  values: PropTypes.array,
+  customClass: PropTypes.string,
+  errorHeader: PropTypes.string,
+  errorDescription: PropTypes.string,
+  keyValue: PropTypes.string,
+  placeholder: PropTypes.string,
+  tabIndex: PropTypes.number,
+  isRequired: PropTypes.bool,
+  prioritySort: PropTypes.bool,
+  clearAfterSelect: PropTypes.bool,
+};
+
+const defaultProps = {
+  bubbledRemovedFilterValue: {
+    filter: '',
+    value: {
+      userId: 0,
+      name: '',
+    },
+  },
+  values: [],
+  placeholder: '',
+  customClass: '',
+  formatter: null,
+  keyValue: 'agency_name',
+  internalValue: ['frec_code', 'cgac_code'],
+  tabIndex: null,
+  isRequired: false,
+  errorHeader: null,
+  errorDescription: null,
+  prioritySort: true,
+  clearAfterSelect: false,
+};
+
+export default class DropdownTypeahead extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.typeahead = null;
+    this.dataDictionary = {};
+    this.internalValueDictionary = {};
+
+    this.state = {
+      dropdownopen: false,
+      value: '',
+      showWarning: false,
+      selectedValues: [],
+      unselectedValues: [],
+    };
+
+    this.onDropdownChange = this.onDropdownChange.bind(this);
+    this.getSelectedName = this.getSelectedName.bind(this);
+    this.passSelectedName = this.passSelectedName.bind(this);
+    this.selectCheckbox = this.selectCheckbox.bind(this);
+    this.unselectCheckbox = this.unselectCheckbox.bind(this);
+    this.removeClickedFilterBarItemFromDropdown = this.removeClickedFilterBarItemFromDropdown.bind(this);
+  }
+
+  componentDidMount() {
+    this.mountAwesomeplete();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.values, this.props.values) && this.typeahead) {
+      this.loadValues();
+    }
+
+    if (!_.isEqual(prevProps.bubbledRemovedFilterValue, this.props.bubbledRemovedFilterValue) && this.props.bubbledRemovedFilterValue.value) {
+      this.removeClickedFilterBarItemFromDropdown(this.props.bubbledRemovedFilterValue);
+    }
+  }
+
+  onDropdownChange() {
+    const currentState = this.state.dropdownopen;
+    this.setState({
+      dropdownopen: !currentState,
+    });
+  }
+
+  getSelectedName() {
+    if (this.state.unselectedValues.includes(this.state.value) || this.state.selectedValues.includes(this.state.value)) {
+      alert('This name is already used, please use the checkbox in the dropdown.');
```

Comment:
We should check with the UX team, but there is probably a better solution than using an alert here.